src/Analysis/Ast/Impl/Extensions/AstExtensions.cs

```diff
@@ -60,9 +62,20 @@ public static Expression FindExpression(this PythonAst ast, SourceLocation locat
             return ast.CommentLocations[match].Column < location.Column;
         }
 
+        internal static bool HasComment(this PythonAst ast, IPythonModule module, string comment, int lineNum) {
+            string line = module.GetLine(ast, lineNum);
+
+            int commentPos = line.IndexOf('#');
+            if (commentPos < 0) {
+                return false; 
+            }
+
+            return line.Substring(commentPos + 1).Contains(comment);
```

Comment:
This function is _still_ doing `Contains`. For `"noqa"`, we need to check the exact string.

I think this should be rewritten to be a function that returns the comment for a line (or `null`), then in `ReportDiagnostics`, you can do `ast.GetComment(SourceSpan.Start.Line)?.Trim().EqualsIgnoreCase("noqa")`.

Reply:
I think it is rarely used case to be an extension. I'd rather have `GetCommentAt` on module which has AST rather than extension to AST that suddenly needs module. AST is a lower layer entity and extensions to it should not require higher level definitions. Consider that AST extensions can actually be placed in Python.Parsing assembly which does not have access to the module.

Reply:
It is called on `module.GetAst()` anyway so module is sufficient